src/package-linker.js

```diff
@@ -393,15 +395,22 @@ export default class PackageLinker {
     }
   }
 
-  determineTopLevelBinLinks(flatTree: HoistManifestTuples): Array<[string, Manifest]> {
+  determineTopLevelBinLinks(
+    flatTree: HoistManifestTuples,
+    workspaceLayout?: WorkspaceLayout,
+  ): Array<[string, Manifest]> {
     const linksToCreate = new Map();
     for (const [dest, {pkg, isDirectRequire}] of flatTree) {
       const {name} = pkg;
 
-      if (!linksToCreate.has(name) || isDirectRequire) {
+      if (
+        !(workspaceLayout && name === workspaceLayout.virtualManifestName) &&
```

Comment:
Nothing prevents it from having a `bin` definition. Also, this is the actual fix after merging from the master. I think the reason is this line: https://github.com/yarnpkg/yarn/pull/4129/commits/7bd20fe7bf4b0d2293940c2416c5eb6de41fbc2c#diff-306a2caea89d886b11134fb770d1ebf6R385